src/core/qfieldcloudprojectsmodel.cpp

```diff
@@ -359,262 +403,596 @@ void QFieldCloudProjectsModel::cancelDownloadProject( const QString &projectId )
+      return;
+    }
+
+    QgsLogger::debug( QStringLiteral( "Project %1: created job with id `%2`" ).arg( projectId ).arg( jobId ) );
+
+    project->jobs[jobType].id = jobId;
 
     emit dataChanged( projectIndex, projectIndex, QVector<int>() << PackagingStatusRole );
 
-    projectGetPackagingStatus( projectId );
+    projectGetJobStatus( projectId, jobType );
   } );
 }
 
-void QFieldCloudProjectsModel::projectGetPackagingStatus( const QString &projectId )
+
+void QFieldCloudProjectsModel::projectGetJobStatus( const QString &projectId, const JobType jobType )
 {
-  const QModelIndex projectIndex = findProjectIndex( projectId );
+  CloudProject *project = findProject( projectId );
 
-  if ( !projectIndex.isValid() )
+  if ( !project )
+  {
+    QgsLogger::debug( QStringLiteral( "Project %1: getting job status, but the project is deleted." ).arg( projectId ) );
     return;
-  CloudProject *project = mProjects[projectIndex.row()];
+  }
 
-  if ( project->packagingStatus == PackagingAbortStatus )
+  if ( project->isPackagingAborted )
+  {
+    QgsLogger::debug( QStringLiteral( "Project %1: getting job status, but project operations are aborted." ).arg( projectId ) );
     return;
+  }
 
-  Q_ASSERT( project->packagingStatus != PackagingUnstartedStatus );
-
-  NetworkReply *downloadStatusReply = mCloudConnection->get( QStringLiteral( "/api/v1/qfield-files/export/%1/" ).arg( projectId ) );
+  if ( !project->jobs.contains( jobType ) )
+  {
+    QgsLogger::debug( QStringLiteral( "Project %1: getting job status, but no package job triggered yet." ).arg( projectId ) );
+    return;
+  }
 
-  project->packagingStatusString = QString();
-  project->apiNetworkReply = downloadStatusReply;
+  QString jobId = project->jobs[jobType].id;
+  QgsLogger::debug( QStringLiteral( "Project %1, job %2: getting job status..." ).arg( projectId, jobId ) );
+  NetworkReply *reply = mCloudConnection->get( QStringLiteral( "/api/v1/jobs/%1/" ).arg( jobId ) );
 
-  QgsLogger::debug( QStringLiteral( "Packaging status requested for \"%1\"" ).arg( projectId ) );
+  connect( reply, &NetworkReply::finished, this, [=]() {
+    reply->deleteLater();
 
-  connect( downloadStatusReply, &NetworkReply::finished, this, [=]() {
-    if ( project->packagingStatus == PackagingAbortStatus )
+    // the project has been deleted
+    if ( !findProject( projectId ) )
+    {
+      QgsLogger::debug( QStringLiteral( "Project %1, job %2: getting job status finished, but the project is deleted." ).arg( projectId, jobId ) );
       return;
+    }
 
-    QNetworkReply *rawReply = downloadStatusReply->reply();
-
-    Q_ASSERT( downloadStatusReply->isFinished() );
-    Q_ASSERT( rawReply );
+    if ( project->isPackagingAborted )
+    {
+      QgsLogger::debug( QStringLiteral( "Project %1, job %2: getting job status finished, but project operations are aborted." ).arg( projectId, jobId ) );
+      return;
```

Comment:
emit projectJobFinished ?

Reply:
I would say I missed them, thanks for spotting. It's would be huge improvement when half of these checks get unnecessary with dedicated CloudProject class.